packages/uicore/src/components/ExpressionDropdown/ExpressionDropdown.tsx

```diff
@@ -159,18 +55,112 @@ export const NewExpressionDropdown = (props: NewExpressionDropdownProps): JSX.El
+          <div className={css.visibleItemTextHeader}>{item.value}</div>
+          {shallShowNesting(item) && (
+            <Icon name={getDropDownIcon(item, isOpen)} className={css.paddingLeft} size={10} />
+          )}
+        </Layout.Horizontal>
+      )
+
+      return (
+        <Popover key={index} minimal>
+          {content}
+          <Menu>
+            {item.children.map(
+              (child, ind) =>
+                child.children.length !== 0 && (
+                  <Menu.Item
+                    key={`${ind} ${child.value}`}
+                    onClick={() => {
+                      dropDownItemClickHandler(child.valueTillHere)
+                      setOpenTargetElement(isOpen, setIsOpen, item.valueTillHere, false)
+                    }}
+                    text={child.value}
+                  />
+                )
+            )}
+          </Menu>
+        </Popover>
+      )
+    }
+  }, [dropDownItemClickHandler, isOpen, setIsOpen])
+
+  const overflowListRenderer = React.useMemo(() => {
+    // eslint-disable-next-line react/display-name
+    return (items: TrieNode[]): JSX.Element => {
+      const clickHandler = () => itemClickHandler(items[0].valueTillHere, true)
+
+      return (
+        <Layout.Horizontal>
+          <Popover isOpen={isPopoverOpen} minimal>
+            <Layout.Horizontal className={css.visibleItemText} onClick={clickHandler}>
+              <div className={css.visibleItemTextHeader}>{items[0].value}</div>
+              <Icon name={getDropDownIcon(items[0], isOpen)} padding={{ left: '5px', right: '10px' }} size={10} />
+            </Layout.Horizontal>
+            <Menu>
+              {items[0].children.map(
+                (child, ind) =>
+                  child.children.length !== 0 && (
+                    <Menu.Item
+                      key={`${ind} ${child.value}`}
+                      onClick={() => {
+                        dropDownItemClickHandler(child.valueTillHere)
+                        setOpenTargetElement(isOpen, setIsOpen, items[0].valueTillHere, false)
+                      }}
+                      text={child.value}
+                    />
+                  )
+              )}
+            </Menu>
+          </Popover>
+          {items.length > 1 ? (
+            <Popover minimal>
+              <div style={{ cursor: 'pointer' }}>...</div>
+              <Menu>
+                {items.map((item, index) => {
+                  if (index !== 0) {
+                    return (
+                      <Menu.Item
+                        key={index}
+                        text={item.value}
+                        onClick={() => dropDownItemClickHandler(item.valueTillHere)}
+                      />
+                    )
+                  }
+                })}
+              </Menu>
+            </Popover>
+          ) : null}
+        </Layout.Horizontal>
+      )
+    }
+  }, [dropDownItemClickHandler, isOpen, setIsOpen])
```

Comment:
`dropDownItemClickHandler` changes every time causing `overflowListRenderer` to change